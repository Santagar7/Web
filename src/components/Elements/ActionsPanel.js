import {useParams} from 'react-router-dom';
import axios from "axios";
axios.defaults.withCredentials = true;

const url = 'http://localhost:7000/';

function ActionsPanel({setActive}){
    const {index} = useParams();
    function CreateNewAction(e){
        e.preventDefault();
        const amount = document.getElementById('amount').value;
        const reason = document.getElementById('reason').value;
        let elem = document.getElementById('message');
        axios.post(url+`actions/${index}`,{
            'amount':amount,
            'reason':reason
        }).then((resp)=>{
            if(resp.data.message==='Success'){
                elem.textContent = 'Success';
                elem.style.color = 'green';
            }
            else{
                elem.textContent = resp.data.message;
                elem.style.color = 'crimson';
            }
        })
    }
    return (
        <div className='panel'>
            <div className="center">
            <h1 id="message">Create new Action</h1>
            <form>
                <input id="amount" type="number" required placeholder="Amount.."/>
                <input id="reason" type="text" required placeholder="Reason/Sourse.."/>
                <button onClick={CreateNewAction}>Save</button>
            </form>
            <form id="cancel">
                <button type="button" className="exit" onClick={()=>{setActive(false)}}>Cancel</button>
            </form>
            </div>
        </div>)
}

export default ActionsPanel;