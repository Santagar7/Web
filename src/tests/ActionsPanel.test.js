import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ActionsPanel from '../components/Elements/ActionsPanel';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><ActionsPanel/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('check create new action', () => {
    render(
        <BrowserRouter>
            <ActionsPanel />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/Create new Action/i);
    expect(linkElement).toBeInTheDocument();
});

test('check actions panel', () => {
    render(
        <BrowserRouter>
            <ActionsPanel />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/Create new Action/i);
    expect(linkElement).toBeInTheDocument();

    const submit = screen.getByText(/Save/i);
    fireEvent.click(submit);

});
