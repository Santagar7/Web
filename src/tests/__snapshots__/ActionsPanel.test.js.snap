// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="panel"
>
  <div
    className="center"
  >
    <h1
      id="message"
    >
      Create new Action
    </h1>
    <form>
      <input
        id="amount"
        placeholder="Amount.."
        required={true}
        type="number"
      />
      <input
        id="reason"
        placeholder="Reason/Sourse.."
        required={true}
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Save
      </button>
    </form>
    <form
      id="cancel"
    >
      <button
        className="exit"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
